platforms/corda-ent/configuration/roles/create/certificates/ambassador/cenm-notary/tasks/nested_main.yaml

```diff
@@ -0,0 +1,229 @@
+  vars:
+    vault_output: "{{ root_certs_notary.stdout | from_yaml }}"
+    type: "rootca"
+    cert_path: "{{ rootca }}"
+  when: root_certs_notary.failed | default(True) == False
+
+############################################################################################
+############################## STORE THE KEYSTORE IN A VARIABLE  ###########################
+
+# Store the exisiting key.jks file in a variable
+- name: Store the existing key.jks file in a variable
+  stat:
+    path: "{{ rootca }}/rootca.jks"
+  register: rootca_stat_result
+
+############################################################################################
+#################### GENERATE ROOT CA CERTIFICATES, IF ABSENT IN VAULT #####################
+
+# Generate Root CA certificates
+- name: Generate Root CA certificates
+  shell: |
+    cd {{ rootca }}
+    eval "keytool -genkey -alias rootca -dname {{ org.subject | quote }} -keystore rootca.jks -storepass cordaent -keypass cordaent"
+    eval "openssl ecparam -name prime256v1 -genkey -noout -out rootca.key"
+    eval "openssl req -x509 -config {{ playbook_dir }}/openssl.conf -new -nodes -key rootca.key -days 1024 -out rootca.pem -extensions v3_ca -subj '/{{ org.subject | regex_replace(',', '/') }}'"
+    eval "openssl pkcs12 -export -name cert -inkey rootca.key -in rootca.pem -out rootcacert.pkcs12 -cacerts -passin pass:'cordaent' -passout pass:'cordaent'"
+    eval "openssl pkcs12 -export -name key -inkey rootca.key -in rootca.pem -out rootcakey.pkcs12 -passin pass:'cordaent' -passout pass:'cordaent'"
+    eval "yes | keytool -importkeystore -srckeystore rootcacert.pkcs12 -srcstoretype PKCS12 -srcstorepass cordaent -destkeystore rootca.jks -deststorepass cordaent"
+    eval "yes | keytool -importkeystore -srckeystore rootcakey.pkcs12 -srcstoretype PKCS12 -srcstorepass cordaent -destkeystore rootca.jks -deststorepass cordaent"
+  when: ( root_certs_cenm.failed | default(False) == True or root_certs_notary.failed | default(False) == True ) and rootca_stat_result.stat.exists == False
+
+############################################################################################
+###################### CHECK IF TLS CERTIFICATES ARE ALREADY IN VAULT ######################
+
+# Check if the ambassador tls certs are already created and stored in the vault for CENM
+- name: Check if the ambassador tls certs are already created and stored in the vault for CENM
+  shell: |
+    vault kv get -format=yaml secret/cenm/{{ peer.name | lower }}/tlscerts
+  environment:
+    VAULT_ADDR: "{{ org.vault.url }}"
+    VAULT_TOKEN: "{{ org.vault.root_token }}"
+  register: ambassador_tls_certs_cenm
+  ignore_errors: yes
+  when: org.type == 'cenm'
+
+# Check if the ambassador tls certs are already created and stored in the vault for notary
+- name: Check if the ambassador tls certs are already created and stored in the vault for notary
+  shell: |
+    vault kv get -format=yaml secret/{{ peer.name }}/tlscerts
```

Comment:
This is fine. But if we are separating cenm and notary, then this code needs to be under create/certificates/ambassador/notary

Reply:
We are seperating notary just for logical seperation (as an organization as we want to make that component HA in future). As per the code, I made the cenm and notary section similar.